packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -1,638 +1,29 @@
-   */
-  public getAll(): Promise<CredentialExchangeRecord[]> {
-    return this.credentialRepository.getAll(this.agentContext)
-  }
-
-  /**
-   * Find a credential record by id
-   *
-   * @param credentialRecordId the credential record id
-   * @returns The credential record or null if not found
-   */
-  public findById(credentialRecordId: string): Promise<CredentialExchangeRecord | null> {
-    return this.credentialRepository.findById(this.agentContext, credentialRecordId)
-  }
-
-  /**
-   * Delete a credential record by id, also calls service to delete from wallet
-   *
-   * @param credentialId the credential record id
-   * @param options the delete credential options for the delete operation
-   */
-  public async deleteById(credentialId: string, options?: DeleteCredentialOptions) {
-    const credentialRecord = await this.getById(credentialId)
-    const service = this.getService(credentialRecord.protocolVersion)
-    return service.delete(this.agentContext, credentialRecord, options)
-  }
-
-  public async findProposalMessage(credentialExchangeId: string): Promise<FindProposalMessageReturn<CSs>> {
-    const service = await this.getServiceForCredentialExchangeId(credentialExchangeId)
-
-    return service.findProposalMessage(this.agentContext, credentialExchangeId)
-  }
-
-  public async findOfferMessage(credentialExchangeId: string): Promise<FindOfferMessageReturn<CSs>> {
-    const service = await this.getServiceForCredentialExchangeId(credentialExchangeId)
-
-    return service.findOfferMessage(this.agentContext, credentialExchangeId)
-  }
-
-  public async findRequestMessage(credentialExchangeId: string): Promise<FindRequestMessageReturn<CSs>> {
-    const service = await this.getServiceForCredentialExchangeId(credentialExchangeId)
-
-    return service.findRequestMessage(this.agentContext, credentialExchangeId)
-  }
-
-  public async findCredentialMessage(credentialExchangeId: string): Promise<FindCredentialMessageReturn<CSs>> {
-    const service = await this.getServiceForCredentialExchangeId(credentialExchangeId)
-
-    return service.findCredentialMessage(this.agentContext, credentialExchangeId)
-  }
-
-  private async getServiceForCredentialExchangeId(credentialExchangeId: string) {
-    const credentialExchangeRecord = await this.getById(credentialExchangeId)
+export class CredentialsModule {
+  public readonly config: CredentialsModuleConfig
 
-    return this.getService(credentialExchangeRecord.protocolVersion)
+  public constructor(config: CredentialsModuleConfigOptions) {
+    this.config = new CredentialsModuleConfig(config)
   }
 
   /**
    * Registers the dependencies of the credentials module on the dependency manager.
    */
-  public static register(dependencyManager: DependencyManager) {
+  public register(dependencyManager: DependencyManager) {
     // Api
-    dependencyManager.registerContextScoped(CredentialsModule)
+    dependencyManager.registerContextScoped(CredentialsApi)
```

Comment:
I am still not the biggest fan of these register function names. I think that `registerContextScoped` is a bit ambiguous and does not tell me when to use this method in the context of AFJ.

aliassing it to something like: `registerPublicApi` or whatever fits better might be worth it to make writing external plugins a lot easier.

Reply:
I see your point. What I'm not sure about is that we're not sure yet what types of components people want to register. Maybe we can come up with a set of components, but it would need to cover all cases. 

- `registerApi` for the api instead of `registerContextScoped`
- `registerStatelessService` instead of `registerSingleton` 
- `registerInstance` stays as is

We would still need another name for the `registerContextScoped` if you have something that is not an API but needs to be unique per context. Do you like this direction better?

Reply:
Yeah this makes it a lot easier to understand. Maybe for the `registerContextScoped` without an api and needs to be unique, we can use `registerStatefulService`? I believe that that is what you want to register, right?

Reply:
I'll raise it during the WG call and gather some more input on this. If we agree we want to change something, I'll raise another PR